### ISSUE TITLE
In File, length of returned values are not checked.

### ISSUE THREAD

**Issue by defuse** [defuse/php-encryption#114]
```
            $ivsize = \openssl_cipher_iv_length($config['CIPHER_METHOD']);
            $iv = \fread($inputHandle, $ivsize);
            if ($iv === false ) {
                throw new Ex\CannotPerformOperation(
                    'Cannot read input file'
                );
            }
```

None of the code below that checks if `$iv` is actually the correct length, so if `EOF` was encountered (or some other weird condition), we'll get a wrong size IV. The same bug probably exists for other parameters read from the file too.

**Comment by paragonie-scott** [comment]
Feel free to copy https://github.com/paragonie/halite/blob/b202f427062c29566b6ae09b1ee72b2bc1746d61/src/File.php#L667-L695 for file reads.

**Comment by paragonie-scott** [comment]
(Aside: I'm going to create a congruent method in Halite for file writes.)

**Comment by paragonie-scott** [comment]
Should be fixed in #129

**Comment by defuse** [comment]
This looks good, except for the issue in #145.
